### ISSUE TITLE
The inconsistency between the good metrics and the poor quality of my prediction visualization.

### ISSUE THREAD

**Issue by TrinhThiBaoAnh** [qubvel/segmentation_models.pytorch#777]
Dear qubvel,

I hope this message finds you well. I recently used your example code "cars segmentation (camvid).ipynb" from your GitHub repository for a segmentation task. While I obtained reasonably good numerical results, I encountered an issue with the visualization of the predicted mask. The object region in the visualization does not appear to be accurate or satisfactory.

I wanted to reach out to you to seek guidance or suggestions on how to address this issue. It would be greatly appreciated if you could provide any insights or recommendations to improve the visual quality of the predicted mask.

Thank you very much for your time and assistance. I look forward to hearing from you.
Best regards,
RedCloudyZz

**Comment by qubvel** [comment]
Hi @TrinhThiBaoAnh, without knowing the task in details I could not advise you much. It could be a code bug, improper metric for your task or something else could went wrong.

**Comment by TrinhThiBaoAnh** [comment]
@qubvel
I have made some changes on your Camvid notebook and I hope that you could review the code to ensure its accuracy.
Here is my modified code: https://drive.google.com/drive/folders/1khmLPjy_G9ebvmOaXBSYOwrjyET83QQP?usp=sharing
Thank you for your attention and support.

**Comment by TrinhThiBaoAnh** [comment]
You can directly access my notebook here: https://drive.google.com/file/d/14YP3topj5dt3v8J0KV5P-U3gmfVgwJiq/view?usp=sharing

**Comment by chefkrym** [comment]
Hi @TrinhThiBaoAnh Tried your code and I would modify my training bit as follows...

`# train model for 40 epochs

max_score = 0

#train accuracy, train loss, val_accuracy, val_loss 
x_epoch_data = []
train_dice_loss = []
train_iou_score = []
valid_dice_loss = []
valid_iou_score = []

for i in range(0, 100):
    
    print('\nEpoch: {}'.format(i))
    train_logs = train_epoch.run(train_loader)
    valid_logs = valid_epoch.run(valid_loader)
    
    x_epoch_data.append(i)
    train_dice_loss.append(train_logs['bce_loss'])
    train_iou_score.append(train_logs['iou_score'])
    valid_dice_loss.append(valid_logs['bce_loss'])
    valid_iou_score.append(valid_logs['iou_score'])
    
    # do something (save model, change lr, etc.)
    if max_score < valid_logs['iou_score']:
        max_score = valid_logs['iou_score']
        torch.save(model, 'E:/UNet_mitb0.pth')
        print('Model saved!')
        
    if i == 25:
        optimizer.param_groups[0]['lr'] = 1e-5
        print('Decrease decoder learning rate to 1e-5!')`

**Comment by TrinhThiBaoAnh** [comment]
I realized that i read masks in wrong way. =)) Thanks for your contribution. @chefkrym
